src/TesApi.Web/BatchScheduler.cs

```diff
@@ -1067,23 +1117,26 @@ private async Task<BatchModels.StartTask> GetStartTaskAsync(string poolId, Batch
 
             if (!dockerConfigured)
             {
-                var packageInstallScript = machineConfiguration.NodeAgentSkuId switch
+                var packageInstallScript = nodeOs switch
                 {
-                    var s when s.StartsWith("batch.node.ubuntu ", StringComparison.OrdinalIgnoreCase) => "echo \"Ubuntu OS detected\"",
-                    var s when s.StartsWith("batch.node.centos ", StringComparison.OrdinalIgnoreCase) => "sudo yum install epel-release -y && sudo yum update -y && sudo yum install -y wget",
+                    NodeOS.Ubuntu => "echo \"Ubuntu OS detected\"",
+                    NodeOS.Centos => "sudo yum install epel-release -y && sudo yum update -y && sudo yum install -y wget",
```

Comment:
agreed

Reply:
This functionality is already present, so removing it in a refactor would be an unexpected behavior change.

Having written that, Centos is going away soon, so this will be removed when support for that OS is removed from Azure.